code/json.sls

```diff
@@ -30,172 +30,232 @@
 
 (library (json)
   (export json-write
-	  json-read)
-  
+          json-read)
+
   (import
-    (except (rnrs) define-record-type error)
-    (rnrs r5rs)
-    (rnrs mutable-pairs)
-    (only (srfi :1 lists) lset-union append-map fold)
-    (srfi :6 basic-string-ports)
-    (srfi :9 records)
-    (srfi :23 error)
-    (only (chezscheme) include)
-    (packrat))
+   (except (rnrs) define-record-type error)
+   (rnrs r5rs)
+   (rnrs mutable-pairs)
+   (only (srfi :1 lists) lset-union append-map fold)
+   (srfi :6 basic-string-ports)
+   (srfi :9 records)
+   (srfi :23 error)
+   (only (chezscheme) include)
+   (packrat))
 
+  ;; write-json :: sexp -> json
+  ;; The interface is three-fold and implemented as a case-lambda
+  ;; 1 argument  = simple unformatted json
+  ;; 2 arguments = first must be 'pretty
+  ;;                 any other symbol will throw an error
+  ;;               second must be the symbolic representation of a scheme object
+  ;; 3 arguments = first must be 'pretty
+  ;;               second must be the symbolic representation of a scheme object
+  ;;               third must be the tabpstop size defined in spaces
   (define json-write
     (let ()
+      ;; control whether or not to pretty print
+      (define pretty? #f)
+      ;; control variables for pretty printing
+      (define indent-level 0)
+      (define tabstop-size 2)
+
+      ;; Print the indentation level
+      (define (display-level p)
+        (display (make-string (* indent-level tabstop-size) #\space) p))
+
```

Comment:
This should probably be rewritten to 
```
(let ((pretty? #f) (indent-level 0) ...)
   ...)
```
rather than `defines`

Reply:
Perhaps keep the helper functions as `define`s, however.

Reply:
You mean at the top level of the `json-write` function? Like in the as-of-yet ignored `()` after let?

I'll try it out.

Reply:
Cool.  Got up the alteration.  Let me know if it is what you meant.